src/Writer.cc

```diff
@@ -0,0 +1,46 @@
+#include "podio/Writer.h"
+
+#include "podio/ROOTWriter.h"
+#if PODIO_ENABLE_RNTUPLE
+  #include "podio/RNTupleWriter.h"
+#endif
+#if PODIO_ENABLE_SIO
+  #include "podio/SIOWriter.h"
+#endif
+
+#include <memory>
+
+namespace podio {
+
+Writer makeWriter(const std::string& filename, const std::string& type) {
+
+  auto endsWith = [](const std::string& str, const std::string& suffix) {
+    return str.size() >= suffix.size() && 0 == str.compare(str.size() - suffix.size(), suffix.size(), suffix);
+  };
+
+  auto lower = [](std::string str) {
+    std::transform(str.begin(), str.end(), str.begin(), [](unsigned char c) { return std::tolower(c); });
+    return str;
+  };
+
+  if ((type == "default" && endsWith(filename, ".root")) || lower(type) == "root") {
+    return Writer{std::make_unique<ROOTWriter>(filename)};
+  }
+  if (lower(type) == "rntuple") {
+#if PODIO_ENABLE_RNTUPLE
+    return Writer{std::make_unique<RNTupleWriter>(filename)};
+#else
+    throw std::runtime_error("ROOT RNTuple writer not available. Please recompile with ROOT RNTuple support.");
+#endif
+  }
+  if ((type == "default" && endsWith(filename, ".sio")) || lower(type) == "sio") {
```

Comment:
I am not entirely sure whether we should check the `type` here as well. currently it would be possible to write `example.root` as an `sio` file, which would then break when trying to read it back, probably with some rather incomprehensible error.

I think it would be less error prone, if we only check the `type` for the `.root` file ending and ignore it entirely for `sio`. Otherwise we would at least also have to foresee the possibility to customize the `type` in the reader interface.

Reply:
Done